python/cuml/test/test_thirdparty.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+
+import numpy as np
+import cupy as cp
+from cuml._thirdparty.sklearn.utils.validation import check_X_y
+
+from cuml._thirdparty.sklearn.utils.extmath import row_norms as cu_row_norms, \
+    _incremental_mean_and_var as cu_incremental_mean_and_var
+from sklearn.utils.extmath import row_norms as sk_row_norms, \
+    _incremental_mean_and_var as sk_incremental_mean_and_var
+
+from cuml._thirdparty.sklearn.utils.sparsefuncs import \
+    inplace_csr_column_scale as cu_inplace_csr_column_scale, \
+    inplace_csr_row_scale as cu_inplace_csr_row_scale, \
+    inplace_column_scale as cu_inplace_column_scale, \
+    mean_variance_axis as cu_mean_variance_axis, \
+    min_max_axis as cu_min_max_axis
+from sklearn.utils.sparsefuncs import \
+    inplace_csr_column_scale as sk_inplace_csr_column_scale, \
+    inplace_csr_row_scale as sk_inplace_csr_row_scale, \
+    inplace_column_scale as sk_inplace_column_scale, \
+    mean_variance_axis as sk_mean_variance_axis, \
+    min_max_axis as sk_min_max_axis
+
+from cuml.test.test_preproc_utils import assert_allclose
+
+
+@pytest.fixture(scope="session")
+def random_dataset(request):
+    X = cp.random.rand(100, 10)
+    return X.get(), X
+
+
+@pytest.fixture(scope="session",
+                params=["cupy-csr", "cupy-csc"])
+def sparse_random_dataset(request):
+    X = cp.random.rand(100, 10)
+    random_loc = cp.random.choice(X.size, int(X.size * 0.3), replace=False)
+    X.ravel()[random_loc] = 0
+    if request.param == 'cupy-csr':
+        X_sparse = cp.sparse.csr_matrix(X)
+    elif request.param == 'cupy-csc':
+        X_sparse = cp.sparse.csc_matrix(X)
+    return X.get(), X, X_sparse.get(), X_sparse
+
+
+def test_check_X_y():
+    X = np.ones((100, 10))
+    y1 = np.ones((100,))
+    y2 = np.ones((100, 1))
+    y3 = np.ones((100, 2))
+    y4 = np.ones((101,))
+
+    check_X_y(X, y1, multi_output=False)
+    check_X_y(X, y2, multi_output=False)
+    with pytest.raises(Exception):
+        check_X_y(X, y3, multi_output=False)
+    with pytest.raises(Exception):
+        check_X_y(X, y4, multi_output=False)
```

Comment:
Do you need a test of multi_output=True working for y4?

Reply:
I'll add it.